### ISSUE TITLE
B::Deparse.pm troublesome expression for infinity

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14380]
Migrated from [rt.perl.org#123526](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123526) (status was 'open')

Searchable as RT123526$

**Comment by p5pRT** [comment]
### From @tromp

Module B&#8203;::Deparse\.pm uses 9\*\*9\*\*9 to denote infinity\.

This causes our Perl code\, which happens to install a floating point
overflow handler through MkSwig'ed C\-code\, to crash\.

Writing 1/0 instead of 9\*\*9\*\*9 avoids this problem\, and would
seem to be the preferred notation for \(positive\) infinity\.

Could we have all uses of 9\*\*9\*\*9 replaced with 1/0 ?

using /usr/local/products/perl/5\.16\.0/lib/4\.16\.0/B/Deparse\.pm
with perl version&#8203;: 5\.16\.0
on OS&#8203;: Linux 3\.11\.10\-21\-default x86\_64

**Comment by p5pRT** [comment]
### From @mauke

Am 31\.12\.2014 um 03&#8203;:14 schrieb John Tromp \(via RT\)&#8203;:
> \# New Ticket Created by  John Tromp
> \# Please include the string&#8203;:  \[perl \#123526\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123526 >
>
>
> Module B&#8203;::Deparse\.pm uses 9\*\*9\*\*9 to denote infinity\.
>
> This causes our Perl code\, which happens to install a floating point
> overflow handler through MkSwig'ed C\-code\, to crash\.
>
> Writing 1/0 instead of 9\*\*9\*\*9 avoids this problem\, and would
> seem to be the preferred notation for \(positive\) infinity\.
>
> Could we have all uses of 9\*\*9\*\*9 replaced with 1/0 ?
>
> using /usr/local/products/perl/5\.16\.0/lib/4\.16\.0/B/Deparse\.pm
> with perl version&#8203;: 5\.16\.0
> on OS&#8203;: Linux 3\.11\.10\-21\-default x86\_64

But 1/0 isn't infinity in Perl; it throws an exception\. Maybe 0 \+ 'inf' 
would be a better choice\.

\-\- 
Lukas Mai \<plokinom@&#8203;gmail\.com>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Wed Dec 31 01&#8203;:01&#8203;:31 2014\, plokinom@&#8203;gmail\.com wrote&#8203;:
> Am 31\.12\.2014 um 03&#8203;:14 schrieb John Tromp \(via RT\)&#8203;:
> > \# New Ticket Created by  John Tromp
> > \# Please include the string&#8203;:  \[perl \#123526\]
> > \# in the subject line of all future correspondence about this issue\.
> > \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123526 >
> >
> >
> > Module B&#8203;::Deparse\.pm uses 9\*\*9\*\*9 to denote infinity\.
> >
> > This causes our Perl code\, which happens to install a floating point
> > overflow handler through MkSwig'ed C\-code\, to crash\.
> >
> > Writing 1/0 instead of 9\*\*9\*\*9 avoids this problem\, and would
> > seem to be the preferred notation for \(positive\) infinity\.
> >
> > Could we have all uses of 9\*\*9\*\*9 replaced with 1/0 ?
> >
> > using /usr/local/products/perl/5\.16\.0/lib/4\.16\.0/B/Deparse\.pm
> > with perl version&#8203;: 5\.16\.0
> > on OS&#8203;: Linux 3\.11\.10\-21\-default x86\_64
> 
> But 1/0 isn't infinity in Perl; it throws an exception\. Maybe 0 \+ 'inf' 
> would be a better choice\.

We could default to 0\+'inf' \(or 9\*\*9\*\*9\) and make it configurable\, just as we make '???' configurable\.

\-\- 

Father Chrysostomos

**Comment by tromp** [comment]
Agreed; 0+'inf' is much less arbitrary, and therefore prefereable to 9**9**9.
